docs/C2SMGuidelines.rst

```diff
@@ -76,13 +73,17 @@ Option 2: spack dev-build
 In order to install software with ``spack dev-build``, one needs a
 local source code.  Spack will then compile the code as it is locally
 present. Contrary to ``spack install``, the version suffix
-(``@master``, ``@v2.7.9``, etc.) does not have any effect on the code version compiled.
-To be safe, always use ``dev-build`` and copy the executable after installation
-into the source folder.
+(``@master``, ``@v2.7.9``, etc.) does not have any effect on the code version compiled,
+since spack will always use the local source.
+Nonetheless it is important to use the correct version suffix i.e ``c2sm-master``
+for all your COSMO-versions building on top of branch ``c2sm-master``. Same is for ``c2sm-features`` etc.
+The reason is that there are some patches spack applies based on the version suffix. Using the 
+wrong version suffix may break your code.
 
 ..  attention::
-    Always store the local sources and the corresponding executable in
-    the same location!
+    You may run into a conflict if you try to install the same spec using ``dev-build``. Spack
+    will tell you this particular spec is already installed. You can circumvent this conflict by keeping your
+    executable alive in your src folder, but uninstall it with Spack using ``spack uninstall <your cosmo spec>``.
```

Comment:
Or by using a different string in `@<string>` for every flavor you build. Like `spack dev-build icon @cpu ...` and `spack dev-build icon @gpu` and `spack dev-build icon @my-new-feature ...`. Then they will all coexist and you can individually `spack load icon @gpu` or uninstall and rebuild them.

Reply:
I added your hint in [332c04c](https://github.com/C2SM/spack-c2sm/pull/645/commits/332c04c518b0ae423cf4127a07845675757ac2ac).